server/datastore/mysql/hosts.go

```diff
@@ -1009,26 +1013,34 @@ func (ds *Datastore) applyHostFilters(ctx context.Context, opt fleet.HostListOpt
 	)
 
 	now := ds.clock.Now()
-	sql, params = filterHostsByStatus(now, sql, opt, params)
-	sql, params = filterHostsByTeam(sql, opt, params)
-	sql, params = filterHostsByPolicy(sql, opt, params)
-	sql, params = filterHostsByMDM(sql, opt, params)
-	sql, params = filterHostsByMacOSSettingsStatus(sql, opt, params)
-	sql, params = filterHostsByMacOSDiskEncryptionStatus(sql, opt, params)
+	sqlStmt, params = filterHostsByStatus(now, sqlStmt, opt, params)
+	sqlStmt, params = filterHostsByTeam(sqlStmt, opt, params)
+	sqlStmt, params = filterHostsByPolicy(sqlStmt, opt, params)
+	sqlStmt, params = filterHostsByMDM(sqlStmt, opt, params)
+	sqlStmt, params = filterHostsByMacOSSettingsStatus(sqlStmt, opt, params)
+	sqlStmt, params = filterHostsByMacOSDiskEncryptionStatus(sqlStmt, opt, params)
 	if enableDiskEncryption, err := ds.getConfigEnableDiskEncryption(ctx, opt.TeamFilter); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return "", nil, ctxerr.Wrap(
+				ctx, &fleet.BadRequestError{
+					Message:     fmt.Sprintf("team is invalid"),
+					InternalErr: err,
+				},
+			)
+		}
```

Comment:
This is the bug fix, where I catch that query based on TeamFilter returned no rows.

Reply:
LGTM, but had a thought: should we return 404 instead of 400? Not necessarily a bad request, just asking for something that doesn't exist. Not a strong opinion tho!

Reply:
Jynx

Reply:
NOOOOO

_sigh_ here's your soda 🥤 @georgekarrv

Reply:
This endpoint tends to return an empty array `[]` when no hosts are found, so returning a 404 didn't seem to fit with existing behavior. 400 seemed to fit better since it also catches invalid negative ids, like `team_id=-1`

Reply:
I'm not as familiar with which should be the right pattern, so I'll let @georgekarrv be the deciding vote. That logic makes sense to me tho @getvictor

Reply:
I'm on the 400 train here, we can add more logic later if we need a 404 to address a specific situation BUT I'm also onboard with a 404 here if we add teamID validation upstream